mibidata/color.py

```diff
@@ -224,3 +225,51 @@ def composite(image, color_map, gamma=1/3, min_scaling=10):
             gamma)
     screened = _screen(data_map)
     return np.uint8(screened * 255)
+
+
+def compose_overlay_from_image_data(image, overlay_settings):
+    """Overlays multiple image channels using overlay_settings from mibitracker.
+
+    Args:
+        image: A MibiImage.
+        overlay_settings: Color overlay dictionary with the following fields:
+            'brightness': float between -1 and 1.
+            'channel': Name of the MibiImage channel.
+            'color': One of the following: 'Cyan', 'Yellow', 'Magenta', 'Green',
+                'Orange', 'Violet', 'Red', 'Blue', or 'Gray'.
+            'intensity_higher': Upper limit of the channel intensity.
+            'intensity_lower': Lower limit of the channel intensity.
+
+    Returns:
+        An NxMx3 uint8 array of an RGB image.
+    """
+    # Convolve with this filter no matter what, to mimic browser rendering.
+    kernel = np.array([
+        [0.05, 0.1, 0.05],
+        [0.1, 0.4, 0.1],
+        [0.05, 0.1, 0.05]
+    ])
+    for i, item in enumerate(overlay_settings):
+        int_array = image[item['channel']]
+        # Because we treat the min differently, don't use np.clip
+        range_min, range_max = item['intensity_lower'], item['intensity_higher']
+        int_array[int_array > range_max] = range_max
+        int_array[int_array < range_min] = 0
+        float_array = int_array / float(range_max)
+        float_array[int_array > 0] += item['brightness']
+        # Apply default filter even if no blurring
+        ndimage.filters.convolve(float_array, kernel, output=float_array)
+        if item['blur'] > 0:
+            ndimage.filters.gaussian_filter(
+                float_array, item['blur'] * 100, output=float_array)
+        np.clip(float_array, 0, 1, out=float_array)
+        if item['color'] == 'Gray':
+            rgb = np.stack((float_array, float_array, float_array), axis=2)
+        else:
+            hsl = _gray2hsl(float_array, COLORS[item['color']])
+            rgb = hsl2rgb(hsl)
+        if i == 0:
+            overlay = rgb
+        else:
+            overlay = _porter_duff_screen(composite, rgb)
```

Comment:
`composite` is the name of another function in this file, I think you meant `overlay`